taskd_services/http/api.py

```diff
@@ -77,22 +81,35 @@ def get_data_path(self, *path) -> str:
     def delete(self):
         env = os.environ.copy()
         env["TASKDDATA"] = TASKD_DATA
-        command = [TASKD_BINARY, "remove", "user", self.org_name, self.user_name]
+        command = [TASKD_BINARY, "remove",
+                   "user", self.org_name, self.user_key]
         delete_proc = subprocess.Popen(
             command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env,
         )
         result = delete_proc.wait()
+
+        del_proc_output = delete_proc.communicate()[
+            0].decode("utf-8").split("\n")
+        result = delete_proc.wait()
         if result != 0:
-            raise TaskdError()
+            raise TaskdError(f'command: {command}, output: {del_proc_output}')
 
         self.deleted = datetime.utcnow()
+
+        db.session.delete(db.session.query(Credential).filter_by(
+            user_key=self.user_key).first())
```

Comment:
Although we delete the actual task _data_ immediately, this model originally relied on soft deletion (see `self.deleted` above) for marking the taskd _account_ record as deleted.  I'm pretty OK with either approach, I suppose, but if we're going to go the direction of deleting that credential record, too, we should probably get rid of the `deleted` datetime field (& things that check that date) so it's not confusing to future readers.